lib/puma/binder.rb

```diff
@@ -229,8 +225,15 @@ def parse(binds, logger, log_msg = 'Listening')
             io = inherit_ssl_listener sock, ctx
             logger.log "* Activated #{str}"
           else
+            ios_len = @ios.length
             io = add_ssl_listener uri.host, uri.port, ctx
-            logger.log "* Listening on #{str}"
+
+            @ios[ios_len..-1].each do |i|
```

Comment:
Is the index necessary? are you just trying to reverse it? just use `reverse_each`

Reply:
No, not reverse.  If there are multiple binds, I believe the current loop will print multiple entries?

Reply:
I think I'm missing some context. What problem does this PR solve?

Reply:
See two tests that had skips removed.  Only pertains to '0.0.0.0' or 'localhost' binds

Reply:
I think I'm following this now. In this specific case, we want to print that we're listening on all newly-added entries to `@ios`.

In that case, I think it might be more straightforward for `add_ssl_listener` to return all instances of `Puma::MiniSSL::Server` it created.

```ruby
ssl_servers = add_ssl_listener uri.host, uri.port, ctx
ssl_servers.each do |server|
  addr = loc_addr_str i
  ssl_qry = str[/\?.+\z/]
  logger.log "* #{log_msg} on ssl://#{addr}#{ssl_qry}"
end
```

Reply:
We can do a similar thing to `add_tcp_listener` to make the code a little easier to read.

Reply:
Either works, but the logging for inherited and/or activated listeners is in the parse methods...

Reply:
I think that's fine. I'm not suggesting we move the logging code (that can stay in `parse`). I'm just suggesting that when `parse` calls `add_ssl_listener`, it gets back the new instances of `Puma::MiniSSL::Server`. That way `parse` can iterate over them easily and log each one.

Reply:
IDK.

I did just notice that `ssl_qry = str[/\?.+\z/]` can be moved out of the loop.

Note that `add_ssl_listener` currently returns the TCP server 'behind' the SSL server.  The SSL server is added to `@ios`.  That may not be needed, as I think it can also change if MiniSSL::Server does a better job of barking like a TCP server.  I think I added MiniSSL::Server#closed? for that purpose in another PR.

A while ago, someone mentioned refactoring/rewriting BInder, so I haven't really looked at it.  I've also got code for abstract unix sockets, and it runs with restart ok...

So, maybe wait on more serious Binder changes with Puma 6?  Just a thought...

Reply:
Yeah, happy to defer a refactor to a future PR 👍

Reply:
I moved the `ssl_qry` statement and also removed the `next unless *Server === i` statements, as with the @ios length handling, they're no longer needed...

Reply:
Both changes look good to me 👍

Reply:
@cjlarose Thanks for looking at it, @nateberkopec also...